packages/experimental/AppearanceAdditions/ios/FRNAppearanceAdditions.m

```diff
@@ -76,18 +80,35 @@ + (BOOL)requiresMainQueueSetup {
     return YES;
 }
 
-RCT_EXPORT_BLOCKING_SYNCHRONOUS_METHOD(horizontalSizeClass)
-{
+/**
+ * When initializing this native module, not all the information required to consistently get all correct initial traits is accessible.
+ * In order to ensure that the correct traits are always be returned the first time they are reuqested, a react tag for a view in that window can
+ * be provided to the native module.
+ *
+ * This initialization step may be needed when accessing traits that can be different in different windows, which include horizontal size class and user interface level.
+ * This initialization step is not necessary if the only traits that are accessed are system wide traits common to all windows, such as  accessibility contrast.
+ */
+RCT_EXPORT_BLOCKING_SYNCHRONOUS_METHOD(initializeTraitCollection:(id)reactTag) {
+    RCTUnsafeExecuteOnMainQueueSync(^{
+        if ([reactTag isKindOfClass:[NSNumber class]]) {
+            UIView *view = [[[self bridge] uiManager] viewForReactTag:reactTag];
+            self->_horizontalSizeClass = RCTHorizontalSizeClassPreference([view traitCollection]);
+            self->_userInterfaceLevel = RCTUserInterfaceLevelPreference([view traitCollection]);
+            self->_accessibilityContrastOption = RCTAccessibilityContrastPreference([view traitCollection]);
+        }
+    });
+    return nil;
```

Comment:
Do we need this?

Reply:
Yeah that's a good question - I don't need to return anything in this method, but RCT_EXPORT_BLOCKING_SYNCHRONOUS_METHOD requires a return value. I thought we would still want the init method to be synchronous on the JS thread so I just returned null, but the other option might be to change it to RCT_EXPORT_METHOD. I can try it and see if it works

Reply:
Just tried RCT_EXPORT_METHOD, and it doesn't work - I think the code here needs to block the JS thread. If there's an alternative to returning nil in RCT_EXPORT_BLOCKING_SYNCHRONOUS_METHOD let me know, but I don't see anything else I could do

Reply:
returning nil is fine, didn't realize it was required.